interop-cats/jvm/src/main/scala/scalaz/zio/interop/stream/Stream.scala

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio.interop.stream
+
+import cats.~>
+import cats.effect.Effect
+import cats.implicits._
+import scalaz.zio.{Chunk, Queue, ZIO, ZManaged, ZSchedule}
+import scalaz.zio.stream.{Take, ZSink, ZStream}
+
+final class Stream[F[+ _], +A] private[stream] (private[stream] val underlying: ZStream[Any, Throwable, A])(
+  implicit liftIO: ZIO[Any, Throwable, ?] ~> F,
+  liftZIO: F ~> ZIO[Any, Throwable, ?],
```

Comment:
I would take the same approach as for STM here: instead of requiring these constraints at construction time, requiring them in the methods; and actually, you only need one constraint: `implicit R: Runtime[???]`. This will summon the whole cats effect hierarchy, so you can go "to" and "from" `F[_]`, as well as use things like `Clock`, if you need them (I saw a `todo` down lower that could benefit from this).

Reply:
Yep, got it, I remove the constraints from the class and added them where I think they are necessary!